doc/lua_api.txt

```diff
@@ -2181,12 +2181,27 @@ Elements
 * Must be used after the `size`, `position`, and `anchor` elements (if present).
 * Disables player:set_formspec_prepend() from applying to this formspec.
 
+### `options[<opt 1>=<val 1>;<opt 2>=<val 2>;...]`
+
+* Sets global options affecting the entire formspec
+* Must be used after the `size`, `position`, `anchor`, and `no_prepend` elements (if present).
+* Possible options:
+    * key_event - boolean: Sets whether to have key events in the field `key_event`.
+      Default false. See [Special Input Events] for more information.
+    * mouse_event: Sets whether to have mouse events in the field `mouse_event`.
+      Possible values:
+        * `none` (default): Mouse events will not be sent.
+        * `all`: Sends all mouse events.
+        * `no_move`: All mouse events will be registered except for `move`; this drastically
+          reduces the amount of mouse events sent, improving performance.
```

Comment:
Then, is there no event ratelimiting at all? That sounds like it'd lead to problems as soon as used anywhere except singleplayer/LAN.
Limiting these two one move event per 200ms would solve this but I'm sure it breaks some usecases people would like (like a paint application, do we want to support that?).

Handling these kinda of events on the server is doomed to always suffer from latency. Perhaps we should <i>wait for SSCSM</i>™ and make this API available to it then.

Reply:
Are you suggesting to limit move events only to 200ms?  That sounds like it wouldn't lead to a lot of trouble (and a paint application sounds like abuse, formspec boxes would be too slow anyway).  Limiting other things like keys or mouse clicks to 200ms could cause trouble since they could be missed very easily just like `player:get_player_control()`.

Also, a (much nicer) event API in CSM will be necessary for the formspec replacement, so SSCSM will have it if it ever comes :)

Reply:
> Are you suggesting to limit move events only to 200ms? 

Yes, only move events. If desired this can even be improved by not delaying sending if the mouse just started/stopped moving.